Digital Mary redesign

joeytakeda opened the pull request:
🎉

Beyond the changes to the style and templates, changes include:

* Adding the thumb width and height to the global vars available to twig
* Adding a query for featured items for the homepage
* Moving the image nav item to below the divider

joeytakeda committed:
Initial work on redesign

joeytakeda committed:
Grid o' images

joeytakeda committed:
Page size to 24 and use the nines template for pagination

joeytakeda committed:
More work on redesign with better image handling and wavy footer

joeytakeda committed:
More work on redesign, finessing menus etc

joeytakeda committed:
Moving images below the line

The image list probably isn't something that we want in the public view since they don't really mean anything without their associated items

joeytakeda committed:
Standard breadcrumb, admin stuff

joeytakeda committed:
Minified

joeytakeda committed:
Merge remote-tracking branch 'upstream/main' into redesign

joeytakeda committed:
Revert "Merge remote-tracking branch 'upstream/main' into redesign"

This reverts commit c9600906152d9ef212f62b4cd7c1318b3b7c2386, reversing
changes made to c72e11593bdf3f4dc0de66c8ab300dfa6585d8b1.

joeytakeda committed:
Fixing conflicts

joeytakeda committed:
More fixes

joeytakeda committed:
Better mobile viewing for the item pages

joeytakeda committed:
More refinements

joeytakeda committed:
Merging and resolving conflicts

joeytakeda committed:
Composer update and small refashioning of templates

joeytakeda committed:
Some small fixes and adding image to below the line menu

joeytakeda committed:
Paginaton work

joeytakeda committed:
Getting a start on the homepage

joeytakeda committed:
More work on index

joeytakeda committed:
Merge remote-tracking branch 'upstream/main' into redesign

Fixing some conflicts too

joeytakeda committed:
Small CSS tweak

joeytakeda committed:
Merging main into redesign

joeytakeda committed:
Merging

joeytakeda committed:
Merge branch 'main' into redesign

joeytakeda committed:
Lots of tweaks, including addressing #21

We've added Athena Ruby now to the repo, but we'll need to acknowledge them still. They don't have a license file included in their downloadable package.

joeytakeda committed:
Lots of work on the various templates, mostly on the items

joeytakeda committed:
More item styling and refactoring how images work

Also I think I've now fixed #21

joeytakeda committed:
Harmonizing templates and getting the homepage looking a bit better

joeytakeda committed:
Adding spellchecking; fixes #24

joeytakeda committed:
Continuing design work

joeytakeda committed:
Moving icons to their own partial

It's a bit inelegant, but a better solution for the time being.

joeytakeda committed:
Adding a featured items query for the home page

joeytakeda committed:
More cleaning up:

* Fixing citation per SH (though, we may still need to figure it out a bit more)
* Adding thumb/width height global vars
* Some documentation/pruning

joeytakeda committed:
Dedup items in locations to make a single list of items

joeytakeda committed:
whitespace and formatting

joeytakeda committed:
Adding tooltip to the viewer thing (first stage for #33)

Rather than add a whole pile of bootstrap JS to the viewer, I've simply
repurposed the title of the image to function as the button tooltip thing.
I think that works well; had to restyle it slightly, but fairly painless, all in all.

joeytakeda committed:
Change featured item query to do 6 images

joeytakeda committed:
Put items for languages

joeytakeda committed:
More work on image viewer and fixing #33

Some small refinements to the image viewer so that the order of icons is a bit more sensible and making it so that the slider isn't unintentionally slid on keypress when the viewer is open.

joeytakeda committed:
Merge branch 'main' into redesign

joeytakeda committed:
Moving recently added items on homepage, which fixes #39

joeytakeda committed:
Fixing thumbnails command merge issue with MJ

joeytakeda committed:
Removing announcements (fixes #40)

joeytakeda committed:
Harmonizing page displays

Changing titles to reflect their content a bit better (i.e. Language: $language) and simplifying the breadcrumb menu creation. It still needs to be properly
handled, but the current solution is a bit more adaptive

joeytakeda committed:
Merge branch 'main' into redesign

Resolved two conflicts by copying contents of main:
* src/Form/LocationType.php
* src/Repository/ItemRepository.php

joeytakeda committed:
Checking src out from main

joeytakeda committed:
Adding a featured items query bit for the home page

joeytakeda committed:
Adding some preliminary handling for remote images (fixes #38)

* Remote images appear on the right hand side like regular images, but they look like buttons (with the external icon thing)--that's easier than doing a whole image for now

* Also removing the add remote image btn in cases where there's already a regular image associated with it

joeytakeda committed:
Fixing up inscription styles and reviving the search partial

joeytakeda committed:
Yarn dependencies including dhilux

joeytakeda committed:
Using the dhilux package

joeytakeda committed:
Move images to below the line

joeytakeda committed:
Adding favicon

joeytakeda committed:
A few more small fixes

joeytakeda committed:
ws

joeytakeda committed:
Fixing admin issues shored up by PR

joeytakeda committed:
Adding .github files and removing superfluous athena ruby stuff

joeytakeda committed:
Taking tests back from main

joeytakeda committed:
Small mod to the explore list on homepage

ubermichael commented on templates/image/partial/list.html.twig:
```diff
@@ -5,7 +5,7 @@
     <div class="row">
         {% for col in 0 .. (cols - 1) %}
-            {% set index = (row * cols + col)|number_format %}
+            {% set index = row * cols + col %}
```
The lack of `|number_format` caused issues in the past. I'm not requesting a change, just pointing out something we need to keep an eye on.

joeytakeda replied:
I didn't do any work on the image index or image_show (since I didn't think it was worth it since those views are just administrative anyway), so I think the lack of number_format is from an old version of the file. I'll update it to use number_format for safety's sake, though.

joeytakeda committed:
Use number_format

Per https://github.com/sfu-dhil/digital-mary/pull/41/files/e9de30115aa14da87ed0c5be99ebf033f32d3047#r562872699
